charts/spire/charts/spire-server/values.yaml

```diff
@@ -175,6 +175,23 @@ ca_subject:
   organization: Example
   common_name: example.org
 
+keyManager:
+  disk:
+    enabled: true
+  awsKMS:
+    enabled: false
+    region: ""
+    # -- Policy to use when creating keys. If no policy is specified, a default policy will be used.
+    keyPolicy:
+      # -- Key policy in JSON format.
+      policy: ""
+      # -- Name of a ConfigMap that has a `policy.json` file with the key policy in JSON format.
+      existingConfigMap: ""
+    # -- Access key ID for the AWS account. If empty, the default credential chain will be used.
+    accessKeyID: ""
+    # -- Secret access key for the AWS account. If empty, the default credential chain will be used.
+    secretAccessKey: ""
```

Comment:
I'm not very familiar with this plugin, but a more secure approach to integrate with AWS is to use a role that can be assumed by adding an annotation on the k8s deployment. This way we don't have to expose secrets like the access keys in the cluster + we can define that key policy on the role.

Reply:
The plugins works with either an access key or assuming a role as you describe. Access key fields are optional.

https://github.com/spiffe/spire/blob/main/doc/plugin_server_keymanager_aws_kms.md#aws-kms-access

Reply:
Seems like the service account needs to be annotated to assume the IAM role. I wonder if it makes sense to add a convenience field here `assumeRoleARN` so users don't need to do it in  `serviceAccount.annotations`.

https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html

Reply:
Could we document the annotation as preferred option. We can keep the accesskeys approach as a fallback for people who can't use a role.

I would prefer to stick with just the annotations.

On the accesskeys we could enhance the documentation to tell people about the assumeRoleARN annotation as preferred approach.

Reply:
Sounds good. I just added an extra sentence to make it clear.

Reply:
@marcofranssen Are we good with these changes?

Reply:
Thanks, this is great, love the inclusion of that link. 🎉